aptos-move/framework/aptos-stdlib/sources/cryptography/ed25519.move

```diff
@@ -122,6 +142,30 @@ module aptos_std::ed25519 {
         signature_verify_strict_internal(signature.bytes, public_key.bytes, bcs::to_bytes(&encoded))
     }
 
+    /// Helper method to construct a SignedMessage struct.
+    public fun new_signed_message<T: drop>(data: T): SignedMessage<T> {
+        SignedMessage {
+            type_info: type_info::type_of<T>(),
+            inner: data,
+        }
+    }
+
+    /// Derives the Aptos-specific authentication key of the given Ed25519 public key.
+    public fun unvalidated_public_key_to_authentication_key(pk: &UnvalidatedPublicKey): vector<u8> {
+        public_key_bytes_to_authentication_key(pk.bytes)
+    }
+
+    /// Derives the Aptos-specific authentication key of the given Ed25519 public key.
+    public fun validated_public_key_to_authentication_key(pk: &ValidatedPublicKey): vector<u8> {
```

Comment:
is this the same as the function `public_key_bytes_to_authentication_key` below?

Reply:
Interface-wise, nope: this takes a reference to a struct.

Maybe it's a good idea not to expose the type-unsafe `public_key_bytes_to_authentication_key` variant though...

Reply:
Okay, ended up not exposing as a way of forcing developers to use the type-safe API.